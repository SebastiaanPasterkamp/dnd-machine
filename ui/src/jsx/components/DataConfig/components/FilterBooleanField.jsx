import React from 'react';
import PropTypes from 'prop-types';

import ControlGroup from '../../ControlGroup';
import InputField from '../../InputField';
import SingleSelect from '../../SingleSelect';

export class FilterBooleanField extends React.Component
{
    items = [
        { id: true, label: "True" },
        { id: false, label: "False" },
    ];

    constructor(props) {
        super(props);
        this.onFieldChange = this.onFieldChange.bind(this);
        this.onFilterChange = this.onFilterChange.bind(this);
    }

    onFieldChange(field) {
        const { setState } = this.props;
        setState({ field });
    }

    onFilterChange(filter) {
        const { setState } = this.props;
        setState({ filter });
    }

    render() {
        const { field, filter } = this.props;

        return (
            <ControlGroup labels={["Field", "Boolean"]}>
                <InputField
                    placeholder="Field..."
                    value={field}
                    type="text"
                    setState={this.onFieldChange}
                />
                <SingleSelect
                    selected={filter}
                    items={this.items}
                    setState={this.onFilterChange}
                />
            </ControlGroup>
        );
    }
};

FilterBooleanField.propTypes = {
    field: PropTypes.string,
    filter: PropTypes.bool,
};

FilterBooleanField.defaultProps = {
    field: '',
    filter: false,
};

export default FilterBooleanField;
