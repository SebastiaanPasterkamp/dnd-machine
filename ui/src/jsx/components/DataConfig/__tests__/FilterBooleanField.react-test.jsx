import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import FilterBooleanField from '../components/FilterBooleanField';

describe('Component: FilterBooleanField', () => {
    const fullProps = {
        field: 'attrib',
        filter: true,
    };

    describe('when rendering', () => {
        const setState = jest.fn();

        it('should work with minimum props', () => {
            const tree = renderer.create(
                <FilterBooleanField
                    setState={setState}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        })

        it('should work with full props', () => {
            const tree = renderer.create(
                <FilterBooleanField
                    {...fullProps}
                    setState={setState}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        })
    });

    it('should emit changes in the field', () => {
        const setState = jest.fn();

        const wrapper = mount(
            <FilterBooleanField
                setState={setState}
            />
        );

        wrapper.find('input[type="text"]').simulate('change', {
            target: { value: fullProps.field }
        });

        expect(setState).toBeCalledWith({
            field: fullProps.field,
        });
    });

    it('should emit changes in the boolean', () => {
        const setState = jest.fn();

        const wrapper = mount(
            <FilterBooleanField
                setState={setState}
            />
        );

        wrapper.find('.fa-angle-down').simulate('click');
        wrapper.find('li[data-value=true]').simulate('click');

        expect(setState).toBeCalledWith({
            filter: true,
        });
    });
});
