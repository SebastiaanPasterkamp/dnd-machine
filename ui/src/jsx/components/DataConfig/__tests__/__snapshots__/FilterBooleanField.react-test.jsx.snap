// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FilterBooleanField when rendering should work with full props 1`] = `
<div
  className="nice-control-group"
>
  <span
    className="nice-input-addon"
  >
    Field
  </span>
  <input
    className="nice-form-control"
    disabled={undefined}
    onChange={[Function]}
    onKeyPress={null}
    placeholder="Field..."
    type="text"
    value="attrib"
  />
  <span
    className="nice-input-addon"
  >
    Boolean
  </span>
  <div
    className="nice-dropdown nice-form-control"
  >
    <button
      className="nice-btn"
      disabled={false}
      onClick={[Function]}
    >
      True
      <i
        className="icon fa-angle-down"
      />
    </button>
    <ul
      className="dropdown-menu"
      onClick={[Function]}
    >
      <li
        className="info"
        data-value={true}
        onClick={[Function]}
      >
        <a>
          True
        </a>
      </li>
      <li
        className={null}
        data-value={false}
        onClick={[Function]}
      >
        <a>
          False
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Component: FilterBooleanField when rendering should work with minimum props 1`] = `
<div
  className="nice-control-group"
>
  <span
    className="nice-input-addon"
  >
    Field
  </span>
  <input
    className="nice-form-control"
    disabled={undefined}
    onChange={[Function]}
    onKeyPress={null}
    placeholder="Field..."
    type="text"
    value=""
  />
  <span
    className="nice-input-addon"
  >
    Boolean
  </span>
  <div
    className="nice-dropdown nice-form-control"
  >
    <button
      className="nice-btn"
      disabled={false}
      onClick={[Function]}
    >
      False
      <i
        className="icon fa-angle-down"
      />
    </button>
    <ul
      className="dropdown-menu"
      onClick={[Function]}
    >
      <li
        className={null}
        data-value={true}
        onClick={[Function]}
      >
        <a>
          True
        </a>
      </li>
      <li
        className="info"
        data-value={false}
        onClick={[Function]}
      >
        <a>
          False
        </a>
      </li>
    </ul>
  </div>
</div>
`;
